library/std/src/sys/pal/uefi/helpers.rs

```diff
@@ -559,3 +526,51 @@ impl Drop for ServiceProtocol {
         }
     }
 }
+
+#[repr(transparent)]
+pub(crate) struct OwnedEvent(NonNull<crate::ffi::c_void>);
+
+impl OwnedEvent {
+    pub(crate) fn new(
+        signal: u32,
+        tpl: efi::Tpl,
+        handler: Option<efi::EventNotify>,
+        context: Option<NonNull<crate::ffi::c_void>>,
+    ) -> io::Result<Self> {
+        let boot_services: NonNull<efi::BootServices> =
+            boot_services().ok_or(BOOT_SERVICES_UNAVAILABLE)?.cast();
+        let mut event: r_efi::efi::Event = crate::ptr::null_mut();
+        let r = unsafe {
+            let create_event = (*boot_services.as_ptr()).create_event;
+            (create_event)(signal, tpl, handler, crate::mem::transmute(context), &mut event)
```

Comment:
Can we avoid the transmute here in favor of something safer?
```rust
context.map(NonNull::as_ptr).unwrap_or(null_mut())
```

Reply:
Is there any reason to. The docs state that `Option<NonNull<T>>` here should be safe to transmute: https://doc.rust-lang.org/std/option/index.html#representation

Reply:
The transmute is valid, yes. But it's a big hammer to use, which could cause a problem if the function is changed in the future. Changes to the `context` type won't result in an error because of the transmute, whereas using the (slightly more verbose) suggestion above will spit out an error if the input type changes.

And in general it's good to avoid unsafe, and reduce the size of unsafe blocks. So you can do `let context = context.map(NonNull::as_ptr).unwrap_or(null_mut())` outside the unsafe block, and then the only unsafe operations are dereferencing boot services and calling `create_event`.